gym_electric_motor/physical_systems/electric_motors.py

```diff
@@ -1716,7 +1716,7 @@ def _flux_limit(self, omega=0, eps_mag=0, u_q_max=0.0, u_rq_max=0.0):
                         l_mr * u_rq_max
             psi_d_max /= - mp['p'] * omega * l_mr
             # clipping flux and setting nominal limit
-            psi_d_max = 0.9 * np.clip(psi_d_max, a_min=None, a_max=mp['l_m'] * i_d)
+            psi_d_max = 0.9 * np.clip(psi_d_max, a_min=0, a_max=np.abs(mp['l_m'] * i_d))

```

Comment:
I_d can be negative and the previously calculated psi_d_max can as well. This should now be caught.